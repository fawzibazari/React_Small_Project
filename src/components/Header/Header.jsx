import { useState } from "react";
import style from "./Header.module.css";

export default function Header(props) {
    let time = new Date().toLocaleTimeString();
    let [ctime, setCTime] = useState();
    const updateTime = () => {
      time = new Date().toLocaleTimeString();
      setCTime(time);
    }
    setInterval(updateTime, 1000);

  
  return (
    <>
    <header data-testid="test_header" className={style.hello}>
      <p className={style.Boss2}>{props.title}</p>
      <div className={style.Boss}><p className={props.text}>{ctime}</p></div>
      </header>
    </>
  );
}
