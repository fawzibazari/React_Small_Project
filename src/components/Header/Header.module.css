.hello {
overflow: hidden;
background-color: #791cf8;
padding: 20px 10px;

}
.Boss {
  float: right;
  color:white;
  background-color: #791cf8; 

  width: 75px;
  border-radius: 23%;
  text-align: center;

}
.Boss:hover {
  float: right;
  color:#791cf8;
  background-color: white; 

  width: 75px;
  border-radius: 23%;
  
}
.Boss2 {
  float: left;
  color: white;
  text-align: center;

}

