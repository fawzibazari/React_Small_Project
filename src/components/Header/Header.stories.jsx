import Header from './Header'
export default {
    title: 'component/Header',
    component : 'Header'
}

export const Normal = () => <Header/>