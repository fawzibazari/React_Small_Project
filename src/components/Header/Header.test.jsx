import { render, screen } from "@testing-library/react";
import Header from "./Header.jsx";

test("print hello world to the user", () => {
    render(<Header/>);

    const hello = screen.getByTestId("test_header");
    expect(hello).toBeInTheDocument();
    expect(hello).toMatchSnapshot();
    expect(hello).toHaveClass("hello");

})