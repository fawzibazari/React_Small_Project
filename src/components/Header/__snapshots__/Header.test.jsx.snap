// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print hello world to the user 1`] = `
<header
  class="hello"
  data-testid="test_header"
>
  <p
    class="Boss2"
  />
  <div
    class="Boss"
  >
    <p />
  </div>
</header>
`;
